vac/raw/decentralized-messaging-ethereum.md

```diff
@@ -0,0 +1,737 @@
+but this would imply using a particular device as receiver and broadcaster within the group. 
+There is an obvious drawback which is having a single device working as a “server”. 
+Should this device be attacked or without connection, there should be a mechanism for its revocation and replacement.
+
+Another approach for communications between devices could be using the keypair of each device. 
+This could open the door to use UPKE, since keypairs should be regenerated frequently.
+
+Each time a device sends a message, either an internal message or an external message, 
+it needs to replicate and broadcast it to all devices in the group.
+
+![Captura de pantalla 2024-03-21 a les 12.07.23.png](https://prod-files-secure.s3.us-west-2.amazonaws.com/1518abd9-c08f-4989-93c1-96525e62bce5/9144c2dd-aa6c-4f7c-9acc-f9dc7f18661d/Captura_de_pantalla_2024-03-21_a_les_12.07.23.png)
+
+The mechanism for the substitution of misbehaving leader devices follows:
+
+1. Each device within a group knows the details of other leader devices.
+This information may come from metadata in received messages, and is replicated by the leader device.
+2. To replace a leader, the user should select any other device within its group and
+use it to send a signed message to all other users.
+3. To get the ability to sign messages,
+this new leader should request the keypair associated to the ID to the wallet.
+4. Once the leader has been changed,
+it revocates access from DCGKA to the former leader using the DCGKA protocol.
+5. The new leader starts a key update in DCGKA.
+
+Not all devices in a group should be able to send messages to other users. 
+Only the leader device should be in charge of sending and receiving messages. 
+To prevent other devices from sending messages outside their group, a requirement should be signing each message. 
+The keys associated to the `ID` should only be in control of the leader device.
+
+The leader device is in charge of setting the keys involved in the DCGKA. 
+This information must be replicated within the group to make sure it is updated.
+
+To detect missing messages or potential misbehavior, messages must include a counter.
+
+### Using UPKE
+
+Managing the group of devices of a user can be done either using a group key protocol such as treeKEM or 
+using the keypair of each device. 
+Setting a common key for a group of devices under the control of the same actor might be excessive, 
+furthermore it may imply some of the problems one can find in the usual setting of a group of different users; 
+for example: one of the devices may not participate in the required updating processes, representing a threat for the group.
+
+The other approach to managing the group of devices is using each device’s keypair, 
+but it would require each device updating these materia frequently, something that may not happens. 
+
+[Updatable public-key encryption](https://eprint.iacr.org/2022/068) is a form of asymetric cryptography 
```

Comment:
```suggestion
[UPKE](https://eprint.iacr.org/2022/068) is a form of asymetric cryptography 
```

Reply:
Or we can change the reference [UPKE](https://eprint.iacr.org/2022/068) as [Updatable public-key encryption](https://eprint.iacr.org/2022/068)